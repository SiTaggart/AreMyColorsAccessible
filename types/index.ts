export interface Levels {
  aa: boolean;
  aaLarge: boolean;
  aaa: boolean;
  aaaLarge: boolean;
}

export interface ColorCombinationTypes {
  accessibility: Levels;
  color: Array<number>;
  contrast: number;
  hex: string;
  model: string;
  valpha: number;
}

export interface ColorCombosTypes {
  color: Array<number>;
  combinations: Array<ColorCombinationTypes>;
  hex: string;
  model: string;
  valpha: number;
  [key: number]: ColorCombosTypes;
}

export interface SiteData {
  background: string;
  textColor: string;
  isLight: boolean;
}
