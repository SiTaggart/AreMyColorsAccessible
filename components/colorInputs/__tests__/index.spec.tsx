/* eslint-env jest */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount, ReactWrapper } from 'enzyme';
import ColorInputs, { ColorInputsProps } from '..';
import ColorCombos from '../../../utils/color-combos';
import { ColorCombosTypes } from '../../../types';

describe('ColorInputs', (): void => {
  let colorCombos: ColorCombosTypes[] | false;
  let mockProps: ColorInputsProps;
  let handleBackgroundColorInputChange: jest.Mock;
  let handleBackgroundColorSliderChange: jest.Mock;
  let handleTextColorInputChange: jest.Mock;
  let handleTextColorSliderChange: jest.Mock;
  let wrapper: ReactWrapper;

  beforeAll(
    (): void => {
      handleBackgroundColorInputChange = jest.fn();
      handleBackgroundColorSliderChange = jest.fn();
      handleTextColorInputChange = jest.fn();
      handleTextColorSliderChange = jest.fn();
    }
  );

  beforeEach(
    (): void => {
      jest.clearAllMocks();
      colorCombos = ColorCombos(['#fff', '#000']);
      if (colorCombos !== false) {
        mockProps = {
          background: '#000',
          colorCombos: colorCombos,
          handleBackgroundColorInputChange: handleBackgroundColorInputChange,
          handleBackgroundColorSliderChange: handleBackgroundColorSliderChange,
          handleTextColorInputChange: handleTextColorInputChange,
          handleTextColorSliderChange: handleTextColorSliderChange,
          isLight: false,
          textColor: '#fff'
        };
      }
      wrapper = mount(<ColorInputs {...mockProps} />);
    }
  );

  it('renders without crashing', (): void => {
    ReactDOM.render(<ColorInputs {...mockProps} />, document.createElement('div'));
    const colorCard = renderer.create(<ColorInputs {...mockProps} />).toJSON();
    expect(colorCard).toMatchSnapshot();
  });

  it('should call the handleTextColorInputChange callback on text color change', (): void => {
    wrapper
      .find('#textColor')
      .at(0)
      .simulate('change', { target: { value: '#ccc' } });
    expect(handleTextColorInputChange).toBeCalledWith('#ccc');
  });

  it('should call the handleBackgroundColorInputChange callback on background color change', (): void => {
    wrapper
      .find('#background')
      .at(0)
      .simulate('change', { target: { value: '#ccc' } });
    expect(handleBackgroundColorInputChange).toBeCalledWith('#ccc');
  });

  it('should call the handleTextColorSliderChange callback on text color slider change', (): void => {
    wrapper
      .find('#textColor-hsl-Lightness')
      .at(0)
      .simulate('input', { target: { value: 12 } });
    expect(handleTextColorSliderChange).toBeCalledWith('#1F1F1F', 'textColor-hsl');
  });

  it('should call the handleBackgroundColorSliderChange callback on background color change', (): void => {
    wrapper
      .find('#background-hsl-Lightness')
      .at(0)
      .simulate('input', { target: { value: '90' } });
    expect(handleBackgroundColorSliderChange).toBeCalledWith('#E6E6E6', 'background-hsl');
  });
});
