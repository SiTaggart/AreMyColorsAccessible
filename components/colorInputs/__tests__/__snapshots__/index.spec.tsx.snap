// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorInputs renders without crashing 1`] = `
.emotion-76 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0 auto;
  max-width: 1200px;
  padding: 0 1rem;
  -webkit-transition: color 400ms ease-out;
  transition: color 400ms ease-out;
}

.emotion-36 {
  padding: 1rem;
  width: 100%;
}

@media (min-width:768px) {
  .emotion-36 {
    -webkit-flex: 1 1 auto;
    -ms-flex: 1 1 auto;
    flex: 1 1 auto;
    width: 50%;
  }
}

.emotion-0 {
  display: inline-block;
  font-size: 1.2rem;
  margin-bottom: 0.5rem;
  width: 100%;
}

.emotion-2 {
  background: rgba(255,255,255,0.1);
  border: solid 1px;
  border-radius: 3px;
  font-size: 1.2rem;
  padding: 0.5rem 1rem;
  margin-bottom: 2rem;
  width: 100%;
}

.emotion-34 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media (min-width:1024px) {
  .emotion-34 {
    margin: 0 -0.5rem;
  }
}

.emotion-12 {
  width: 100%;
}

@media (min-width:1024px) {
  .emotion-12 {
    -webkit-flex: 1 0 auto;
    -ms-flex: 1 0 auto;
    flex: 1 0 auto;
    padding: 0 0.5rem;
    width: 33.3333%;
  }
}

.emotion-8 {
  -webkit-appearance: none;
  background: transparent;
  border-radius: 2px;
  color: currentColor;
  display: block;
  margin: 10px 0;
  width: 100%;
}

.emotion-8::-webkit-slider-runnable-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-8::-webkit-slider-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
  -webkit-appearance: none;
  margin-top: -9px;
}

.emotion-8:focus::-webkit-slider-runnable-track {
  background: currentColor;
}

.emotion-8::-moz-range-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px;
}

.emotion-8::-moz-range-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-8::-ms-track {
  width: 100%;
  height: 4px;
  cursor: pointer;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  background: transparent;
  border-color: transparent;
  border-width: 8px 0;
  color: transparent;
}

.emotion-8::-ms-fill-lower {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-8::-ms-fill-upper {
  box-shadow: 0 0 0 #fff,0 0 0 #fff;
  background: currentColor;
  border: 1px solid currentColor;
  border-radius: 5px * 2;
}

.emotion-8::-ms-thumb {
  box-shadow: 1px 1px 1px #999,0 0 1px #a6a6a6;
  border: 1px solid currentColor;
  height: 20px;
  width: 8px;
  border-radius: 2px;
  background: currentColor;
  cursor: pointer;
}

.emotion-8:focus::-ms-fill-lower {
  background: currentColor;
}

.emotion-8:focus::-ms-fill-upper {
  background: currentColor;
}

<div
  className="emotion-76 emotion-77"
  data-test="color-input-form"
  style={
    Object {
      "color": "#fff",
    }
  }
>
  <div
    className="emotion-36 emotion-37"
  >
    <label
      className="emotion-0 emotion-1"
      htmlFor="textColor"
    >
      Text Color
    </label>
    <input
      autoComplete="off"
      className="emotion-2 emotion-3"
      id="textColor"
      onChange={[Function]}
      style={
        Object {
          "borderColor": "#fff",
          "color": "inherit",
        }
      }
      type="text"
      value="#fff"
    />
    <div
      className="emotion-34 emotion-35"
      data-test="form-hsl-sliders"
    >
      <div
        className="emotion-12 emotion-13"
      >
        <div
          className="emotion-6 emotion-7"
        >
          <label
            className="emotion-0 emotion-1"
            htmlFor="textColor-hsl-Hue"
          >
            Hue 0°
          </label>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <input
            className="emotion-8 emotion-9"
            id="textColor-hsl-Hue"
            max={360}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            type="range"
            value={0}
          />
        </div>
      </div>
      <div
        className="emotion-12 emotion-13"
      >
        <div
          className="emotion-6 emotion-7"
        >
          <label
            className="emotion-0 emotion-1"
            htmlFor="textColor-hsl-Saturation"
          >
            Saturation 0%
          </label>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <input
            className="emotion-8 emotion-9"
            id="textColor-hsl-Saturation"
            max={100}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            type="range"
            value={0}
          />
        </div>
      </div>
      <div
        className="emotion-12 emotion-13"
      >
        <div
          className="emotion-6 emotion-7"
        >
          <label
            className="emotion-0 emotion-1"
            htmlFor="textColor-hsl-Lightness"
          >
            Lightness 100%
          </label>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <input
            className="emotion-8 emotion-9"
            id="textColor-hsl-Lightness"
            max={100}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            type="range"
            value={100}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="emotion-36 emotion-37"
  >
    <label
      className="emotion-0 emotion-1"
      htmlFor="background"
    >
      Background
    </label>
    <input
      autoComplete="off"
      className="emotion-2 emotion-3"
      id="background"
      onChange={[Function]}
      style={
        Object {
          "borderColor": "#fff",
          "color": "inherit",
        }
      }
      type="text"
      value="#000"
    />
    <div
      className="emotion-34 emotion-35"
      data-test="form-hsl-sliders"
    >
      <div
        className="emotion-12 emotion-13"
      >
        <div
          className="emotion-6 emotion-7"
        >
          <label
            className="emotion-0 emotion-1"
            htmlFor="background-hsl-Hue"
          >
            Hue 0°
          </label>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <input
            className="emotion-8 emotion-9"
            id="background-hsl-Hue"
            max={360}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            type="range"
            value={0}
          />
        </div>
      </div>
      <div
        className="emotion-12 emotion-13"
      >
        <div
          className="emotion-6 emotion-7"
        >
          <label
            className="emotion-0 emotion-1"
            htmlFor="background-hsl-Saturation"
          >
            Saturation 0%
          </label>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <input
            className="emotion-8 emotion-9"
            id="background-hsl-Saturation"
            max={100}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            type="range"
            value={0}
          />
        </div>
      </div>
      <div
        className="emotion-12 emotion-13"
      >
        <div
          className="emotion-6 emotion-7"
        >
          <label
            className="emotion-0 emotion-1"
            htmlFor="background-hsl-Lightness"
          >
            Lightness 0%
          </label>
        </div>
        <div
          className="emotion-10 emotion-11"
        >
          <input
            className="emotion-8 emotion-9"
            id="background-hsl-Lightness"
            max={100}
            min={0}
            onChange={[Function]}
            onInput={[Function]}
            type="range"
            value={0}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
